File: 22.iterator/iterator.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

// Colección
class ProductCollection implements Iterable<String> {
    private List<String> products = new ArrayList<>();

    public void addProduct(String product) {
        products.add(product);
    }

    @Override
    public Iterator<String> iterator() {
        return products.iterator();
    }
}

// Uso
public class iterator {
    public static void main(String[] args) {
        ProductCollection collection = new ProductCollection();
        collection.addProduct("Laptop");
        collection.addProduct("Smartphone");
        collection.addProduct("Tablet");

        for (String product : collection) {
            System.out.println(product);
        }
    }
}

File: 22.iterator/iterator.py
# Colección
class ProductCollection:
    def __init__(self):
        self.products = []

    def add_product(self, product):
        self.products.append(product)

    def __iter__(self):
        return iter(self.products)

# Uso
collection = ProductCollection()
collection.add_product("Laptop")
collection.add_product("Smartphone")
collection.add_product("Tablet")

for product in collection:
    print(product)